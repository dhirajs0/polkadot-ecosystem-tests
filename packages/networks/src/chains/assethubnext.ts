import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNext: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNext) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNext = defineChain({
  name: 'assetHubNext',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNext,
  initStorages: getInitStorages(custom.assetHubNext),
})
