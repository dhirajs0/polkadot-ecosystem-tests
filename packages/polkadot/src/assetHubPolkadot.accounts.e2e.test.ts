import { assetHubPolkadot } from '@e2e-test/networks/chains'
import { registerTestTree, transferFunctionsTests } from '@e2e-test/shared'

registerTestTree(
  transferFunctionsTests(assetHubPolkadot, {
    testSuiteName: 'Polkadot Asset Hub Accounts',
    addressEncoding: 0,
  }),
)
