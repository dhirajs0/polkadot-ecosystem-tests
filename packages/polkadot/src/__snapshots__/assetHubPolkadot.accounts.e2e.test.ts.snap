// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Polkadot Asset Hub Accounts > \`force_transfer\` > force transfer below ED can kill source account > events of \`force_transfer\` from Alice to Bob 1`] = `
[
  {
    "data": {
      "amount": "(rounded 12000)",
      "from": "13UVJyLgBASGhE2ok3TvxUfaQBGUt88JCcdYjHvUhvQkFTTx",
      "to": "12ixt2xmCJKuLXjM3gh1SY7C3aj4gBoBUqExTBTGhLCSATFw",
    },
    "method": "Transfer",
    "section": "balances",
  },
  {
    "data": {
      "account": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
    },
    "method": "KilledAccount",
    "section": "system",
  },
  {
    "data": {
      "account": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
      "amount": "(rounded 33000000)",
    },
    "method": "DustLost",
    "section": "balances",
  },
  {
    "data": {
      "account": "13Mbv5CRZeSZUqgN4fnS7bQ8XekARHe4kHaekM9mC3J4Fh7e",
    },
    "method": "NewAccount",
    "section": "system",
  },
  {
    "data": {
      "account": "13Mbv5CRZeSZUqgN4fnS7bQ8XekARHe4kHaekM9mC3J4Fh7e",
      "freeBalance": 100000000,
    },
    "method": "Endowed",
    "section": "balances",
  },
  {
    "data": {
      "amount": 100000000,
      "from": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
      "to": "13Mbv5CRZeSZUqgN4fnS7bQ8XekARHe4kHaekM9mC3J4Fh7e",
    },
    "method": "Transfer",
    "section": "balances",
  },
]
`;

exports[`Polkadot Asset Hub Accounts > transfer_allow_death > should allow killing sender account > events when Alice \`transfer_allow_death\` to Bob 1`] = `
[
  {
    "data": {
      "amount": "(rounded 15000000)",
      "who": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {
      "account": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
    },
    "method": "KilledAccount",
    "section": "system",
  },
  {
    "data": {
      "account": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
      "amount": "(rounded 19000000)",
    },
    "method": "DustLost",
    "section": "balances",
  },
  {
    "data": {
      "account": "13Mbv5CRZeSZUqgN4fnS7bQ8XekARHe4kHaekM9mC3J4Fh7e",
    },
    "method": "NewAccount",
    "section": "system",
  },
  {
    "data": {
      "account": "13Mbv5CRZeSZUqgN4fnS7bQ8XekARHe4kHaekM9mC3J4Fh7e",
      "freeBalance": 100000000,
    },
    "method": "Endowed",
    "section": "balances",
  },
  {
    "data": {
      "amount": 100000000,
      "from": "1ZLR63GQkKph3XmxzziZ1RrPsL6CrJJ1izhgR1mzMaoHZJA",
      "to": "13Mbv5CRZeSZUqgN4fnS7bQ8XekARHe4kHaekM9mC3J4Fh7e",
    },
    "method": "Transfer",
    "section": "balances",
  },
]
`;

exports[`Polkadot Asset Hub Accounts > transfer_allow_death > transfer of some funds does not kill sender account > events when Alice transfers 1 ED to Bob with sufficient balance 1`] = `
[
  {
    "data": {
      "amount": "(rounded 15000000)",
      "who": "14iZ76XKqvAgLe4LLRRs6L93BE5779pSNQSWvkX1zXG4YvZ4",
    },
    "method": "Withdraw",
    "section": "balances",
  },
  {
    "data": {
      "account": "12EqLbWZe5jLsfgESG7L7kaNm2fDjKRpSh1AXu9bQdN2k6vH",
    },
    "method": "NewAccount",
    "section": "system",
  },
  {
    "data": {
      "account": "12EqLbWZe5jLsfgESG7L7kaNm2fDjKRpSh1AXu9bQdN2k6vH",
      "freeBalance": 100000000,
    },
    "method": "Endowed",
    "section": "balances",
  },
  {
    "data": {
      "amount": 100000000,
      "from": "14iZ76XKqvAgLe4LLRRs6L93BE5779pSNQSWvkX1zXG4YvZ4",
      "to": "12EqLbWZe5jLsfgESG7L7kaNm2fDjKRpSh1AXu9bQdN2k6vH",
    },
    "method": "Transfer",
    "section": "balances",
  },
]
`;
